Backend/api/user/router.js

```diff
@@ -5,31 +5,48 @@ const {
   checkAdminAuth,
 } = require("../middleware/auth-validation");
 const {
+  validationResults,
   isUserValid,
-  valdiationResults,
+  isWorkoutValid,
 } = require("../middleware/body-validation");
 
 const router = require("express").Router();
 
-router.get("/:id?", checkUserAuth, userController.getUser);
+router.get("/get/:_id?", checkUserAuth, userController.getUser);
 
 router.post(
   "/register",
   isUserValid,
-  valdiationResults,
+  validationResults,
   userController.register
 );
 
 router.post("/login", userController.login);
 
 router.put(
-  "/:_id",
+  "/update/:_id?",
   checkUserAuth,
   isUserValid,
-  valdiationResults,
+  validationResults,
   userController.updateUser
 );
 
 router.delete("/:_id", checkAdminAuth, userController.deleteUser);
 
+router.get("/getWorkouts", checkUserAuth, userController.getWorkouts);
+
+router.put(
```

Comment:
This creates a resource without knowing its id beforehand. A POST method is much more appropriate.